### PR TITLE
Add footer, matching functionality of header

### PR DESCRIPTION
**Pull request by cgarwood**
Adds a global footer, with functionality that matches the header.

Looks like vscode or the pre-commit hook re-formatted a bunch of `main.less` 🙁

Haven't contributed to this project in a few years (the pre-rollup days) so please let me know if anything needs adjusted.

### CODE REVIEW

**Comment by rchl** [comment]
I've extracted your formatting changes and committed as f0ccc11ea3fd25a93918bc0ecb75c9f5e4347d7b
Could you rebase your changes now?

**Comment by rchl** [comment]
This needs to be documented in the readme. Otherwise the feature doesn't exist to anyone else. :)

File: scripts/controllers/main.js [hunk 1]
```diff
@@ -876,6 +876,14 @@ App.controller('Main', function ($scope, $timeout, $location, Api, tmhDynamicLoc
       return page.header;
    };
 
+   $scope.getFooter = function (page) {
```
Comment (rchl) [inline scripts/controllers/main.js:879]: It's never called with `page` argument. Does it make sense to support it per-page like `getFooter` or not?

File: styles/main.less [hunk 2]
```diff
@@ -311,10 +411,13 @@ body {
...
          margin-top: 0.1em;
          font-size: 1em;
-
       }
    }
 }
+.footer {
+   bottom: 0px;
+   top: auto;
+}
```
Comment (rchl) [inline styles/main.less:420]: We'd need this to also work correctly with `menuPosition: MENU_POSITIONS.BOTTOM`

File: scripts/controllers/main.js (hunk 1 above)
Comment (cgarwood) [reply scripts/controllers/main.js:879]: I don't use it per-page but if headers can be per-page i think it makes sense for this to be per-page capable too

**Comment by cgarwood** [comment]
Thanks for the review! I think all the comments are addressed now 🙂

File: styles/main.less [hunk 3]
```diff
@@ -414,6 +415,15 @@ body {
    }
 }
 
+.footer {
+   bottom: 0px;
+   top: auto;
+}
+
+.-menu-bottom .footer {
+   bottom: 62px;
```
[the remark this answers is not in the document]
Comment (cgarwood) [reply styles/main.less:424]: This could prove to be difficult. `@pageBottomPadding` puts it pretty far above the menu bar. `@pageBottomPaddingBig` and `@pageBottomPaddingSmall` put it in the right spot on `CUSTOM_THEMES.COMPACT` but too low on WINPHONE and HOMEKIT.

I'll do some mroe digging, may have to modify the styles for it on each theme.

File: styles/themes.less [hunk 4]
```diff
@@ -9,671 +9,695 @@
 @pageMenuIconVerticalMarginSmall: 3px;
 @pageMenuIconVerticalMarginBig: 5px;
 
-@pageBottomPaddingSmall: 2 * @pageMenuBottomPadding + @pageMenuIconSizeSmall
-      + 2 * @pageMenuIconVerticalMarginSmall;
-@pageBottomPaddingBig: 2 * @pageMenuBottomPadding + @pageMenuIconSizeBig
-      + 2 * @pageMenuIconVerticalMarginBig;
-
+@pageBottomPaddingSmall: 2 * @pageMenuBottomPadding + @pageMenuIconSizeSmall + 2 * @pageMenuIconVerticalMarginSmall;
+@pageBottomPaddingBig: 2 * @pageMenuBottomPadding + @pageMenuIconSizeBig + 2 * @pageMenuIconVerticalMarginBig;
```
Comment (rchl) [inline styles/themes.less:13]: I'm fine with reformatting but it has to be done in a separate PR. I need to be able to see what has actually changed.

Comment (cgarwood) [reply styles/themes.less:13]: Gah, sorry about that. I didn't realize vs-code auto reformatted it again :( - will fix that.

**Review by rchl — APPROVED** [review]

**Comment by rchl** [comment]
Sorry about the delay, it feel off my radar.

Will merge and release new version ASAP.

**Comment by cgarwood** [comment]
Thanks!
